plugins/modules/vmware_guest_controller.py

```diff
@@ -302,13 +309,13 @@ def check_ctl_disk_exist(self, ctl_type=None, bus_number=None):
 
         return ctl_specified, disks_attached_exist
 
-    def create_controller(self, ctl_type, bus_number=0):
+    def create_controller(self, ctl_type, bus_number=0, bus_sharing='noSharing'):
```

Comment:
If I understand your code correct, this implements an _implicit_ default for `bus_sharing`. I think it would be better to make this default explicit, that is defining a `default='noSharing'` in the `argument_spec` for the `bus_sharing` parameter. And, of course, mentioning this in the documentation.